service-locator/core/src/main/scala/com/lightbend/lagom/internal/client/ServiceNameMapper.scala

```diff
@@ -17,95 +17,56 @@ import scala.collection.JavaConverters._
 private[lagom] class ServiceNameMapper(config: Config) {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private val serviceLookupRegex = Some(config.getString("service-lookup-regex"))
-    .filter(_.nonEmpty)
-    .map(Pattern.compile)
-
   private val defaultPortName = Some(config.getString("defaults.port-name")).filter(_.nonEmpty)
   private val defaultPortProtocol = Some(config.getString("defaults.port-protocol")).filter(_.nonEmpty)
-  private val serviceNameSuffix = config.getString("service-name-suffix")
+  private val defaultScheme = Some(config.getString("defaults.scheme")).filter(_.nonEmpty)
 
-  private val portNameSchemeMapping = {
-    val mappings = config.getConfig("port-name-scheme-mapping")
+  private val serviceLookupMapping: Map[String, ServiceLookup] =
     config
-      .getObject("port-name-scheme-mapping")
+      .getObject("service-name-mappings")
+      .entrySet()
       .asScala
-      .map {
-        case (key, _) => key -> mappings.getString(key)
+      .map { entry =>
+        if (entry.getValue.valueType != ConfigValueType.OBJECT) {
+          throw new IllegalArgumentException(
+            s"Illegal value type in service-name-mappings: ${entry.getKey} - ${entry.getValue.valueType}")
+        }
+        val configEntry = entry.getValue.asInstanceOf[ConfigObject].toConfig
+
+        def getOptionString(name: String) =
+          if (configEntry.hasPath(name)) Some(configEntry.getString(name).trim)
```

Comment:
`hasPath` checks whether a value is present and non-null. And there is also `hasPathOrNull` which checks whether a value is present at the given path, even if the value is null.

Just confirming that what we want here is the first behavior (meaning `Some(null)` has no semantics).

Reply:
Correct, `Some(null)` has not semantics. However, we do allow the user to unset the scheme by doing `defaults.scheme = ""`. Maybe we should also allow it to be `null` and in that case, better to choose for `Option(configEntry.getString(name))`.